test/python/test_localjob.py

```diff
@@ -111,6 +120,15 @@ def test_run_async(self):
                                    '{0} s'.format(timeout))
             time.sleep(1)
 
+        # Wait for all the jobs to finish.
+        # TODO: this causes the test to wait until the 15 qubit jobs are
+        # finished, which might take long (hence the @slow_test). Waiting for
+        # the result is needed as otherwise the jobs would still be running
+        # once the test is completed, causing failures in subsequent tests as
+        # the executor's queue might be overloaded.
```

Comment:
Perhaps we could use [`shutdown`](https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.Executor.shutdown) to free the resources owned by the executor without delaying the tests but this also works for me.

Reply:
Thanks @delapuente - it would be worth exploring indeed, as it might improve things a bit! In general I'd like to explore other alternatives that end up in less demanding jobs in this test, hopefully in a more comprehensive PR :+1: